n3fit/src/evolven3fit_new/evolve.py

```diff
@@ -0,0 +1,221 @@
+import pathlib
+import logging
+import sys
+
+import numpy as np
+from reportengine.compat import yaml
+
+from ekobox import genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import output
+from validphys.api import API
+
+from . import utils, eko_utils
+
+
+log = logging.getLogger(__name__)
```

Comment:
Change name, later on you are using `log` as the variable to iterate over, and you are overwriting it.

It makes sense that you use the short name for the inner variable (the one of the for loop), since there are few instructions, and a longer one for objects available module level, to prevent confusion and accidental overwriting.

I'd recommend `_logger`, with the leading underscore, since this variable is imported with the module, but it should be used only internally.

```suggestion
_logger = logging.getLogger(__name__)
```

Reply:
Yes you are right, thanks to have spotted this and sorry for the error. I am solving this now.